patches/ios-chrome-browser-flags-BUILD.gn.patch

```diff
@@ -0,0 +1,12 @@
+diff --git a/ios/chrome/browser/flags/BUILD.gn b/ios/chrome/browser/flags/BUILD.gn
+index c21300755d99b35904352abb7fde533048bd29fe..3091c2a2615e661e004060e81718e34769d0428d 100644
+--- a/ios/chrome/browser/flags/BUILD.gn
++++ b/ios/chrome/browser/flags/BUILD.gn
+@@ -105,6 +105,7 @@ source_set("flags") {
+   if (ios_enable_screen_time) {
+     deps += [ "//ios/chrome/browser/screen_time/model:feature_flags" ]
+   }
++  deps += [ "//brave/components/brave_rewards/common/buildflags" ]
```

Comment:
why do we have a buildflags dependency here? Also this patch does not follow patching standards. It should be `deps += brave_flags_deps`
Same below, the patch is incorrect and I don't understand why we would have a p3a buildflag in prefs. I think there's likely a problem with the patches we used here because these things seem misplaced to me

Reply:
Hmm, so this seems strange. We have:
1. `brave/ios/browser/flags/about_flags.mm`, where we define `BRAVE_ABOUT_FLAGS_FEATURE_ENTRIES`
2. `brave/chromium_src/ios/chrome/browser/flags/about_flags.mm` which #include the above
3. `brave/patches/ios-chrome-browser-flags-about_flags.mm.patch` that patches in `BRAVE_ABOUT_FLAGS_FEATURE_ENTRIES`

These all seem right, but then we also have:
4. `brave/chromium_src/ios/chrome/browser/flags/BUILD.gn` which I think was meant to be depended on by the upstream?
5. `brave/ios/browser/flags/BUILD.gn` which has a source_set to build [1.] and all necessary deps, even though the source is already built by the upstream `/ios/browser/flags` target.

cc: @kylehickinson

Reply:
The iOS implementation mirrors desktop/android, but the BUILD.gn differed a bit because desktop's `about_flags.cc`'s source_set belongs to `//chrome/browser` which we have a sources.gni patch into: https://github.com/brave/brave-core/blob/745f2e54a36a2316e0b90f6e0fb2f4755e149a0c/patches/chrome-browser-BUILD.gn.patch

I think I set up [4] to ensure the dep existed for `brave/ios/browser/flags/about_flags.mm` when it compiles the chromium_src override since I include both mm files in it. Not entirely sure why these patches would be needed though, indeed it does seem like there is some other issue at play here

Reply:
the only thing that is in that file is `ENABLE_GEMINI_WALLET` which is not enabled on ios. I think we can safely remove this until/if we implement gemini wallet on ios because the feature flag doesn't make sense for a feature that doesn't exist anyway

Reply:
what about p3a in shared prefs below?

Reply:
I think it's the same issue, `brave/chromium_src/ios/chrome/browser/shared/model/prefs/browser_prefs.mm` has all of its deps listed in `brave/chromium_src/ios/chrome/browser/shared/model/prefs/BUILD.gn`, but the upstream `ios/chrome/browser/shared/model/prefs/BUILD.gn` doesn't have a dependency on it.

Reply:
I think all the deps need to go to the respective sources.gni and patched into the upstream `BUILD.gn`s

Reply:
I guess I just didn't expect to find `RegisterBrowserStatePrefs` in a directory called `shared/model/prefs`, but that one seems correct other than the format of the patch should be `deps += brave_browser_prefs_deps`

Reply:
but this one can be removed along with the feature flag that doesn't apply to ios

Reply:
Re `patches/ios-chrome-browser-flags-BUILD.gn.patch`: shouldn't `ios/chrome/browser/flags` have all the dependencies that `brave/chromium_src/ios/chrome/browser/flags/about_flags.mm` introduces? It's not just `//brave/components/brave_rewards/common/buildflags` that @cdesouza-chromium added, it should be all the deps from `//brave/ios/browser/flags`, no?

Reply:
I think https://github.com/brave/brave-core/pull/24966/commits/8ccce7fb701f47eacd8578c3ca1fc7a288c9f31b should be the fix for the flags deps.

Reply:
Similarly, 621ec6330bf4b77fa8f9bc402e4f1ef9391951e8 for ios/chrome/browser/shared/model/prefs

Reply:
@mkarolin the fix for the flag deps is to remove the feature flag that doesn't apply to ios. The prefs one is ok.